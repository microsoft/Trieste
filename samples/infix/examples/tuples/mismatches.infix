foo = (1, 2, 3);

// every one of these should error in the calc stage due to type mismatch
print "tpl +" foo + 1;
print "tpl -" foo - 1;
print "tpl *" foo * 1;
print "tpl /" foo / 1;

// wrong way round... (should just report invalid types though)
print "bad idx" 1 . foo;

print "nested bad" (42, foo + 1);
