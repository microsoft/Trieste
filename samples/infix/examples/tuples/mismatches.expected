//! --enable-tuples --tuples-require-parens --expect-fail calculate
//! --enable-tuples --tuples-require-parens --use-parser-tuples --expect-fail calculate
//! --enable-tuples --expect-fail calculate
Invalid maths op
-- mismatches.infix:3:14
print "tpl +" foo + 1;
              ~~~~~~~

Invalid maths op
-- mismatches.infix:4:14
print "tpl -" foo - 1;
              ~~~~~~~

Invalid maths op
-- mismatches.infix:5:14
print "tpl *" foo * 1;
              ~~~~~~~

Invalid maths op
-- mismatches.infix:6:14
print "tpl /" foo / 1;
              ~~~~~~~

Invalid or unevaluatable tuple index
-- mismatches.infix:9:16
print "bad idx" 1 . foo;
                ~~~~~~~

Invalid maths op
-- mismatches.infix:11:24
print "nested bad" (42, foo + 1);
                        ~~~~~~~
